src/main/java/com/purbon/kafka/topology/api/ksql/KsqlApiClient.java

```diff
@@ -11,38 +12,48 @@
 import io.confluent.ksql.api.client.ClientOptions;
 import io.confluent.ksql.api.client.StreamInfo;
 import io.confluent.ksql.api.client.TableInfo;
+
+import javax.annotation.Nullable;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.net.URL;
+import java.util.*;
 import java.util.concurrent.ExecutionException;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class KsqlApiClient implements ArtefactClient {
 
-  private String server;
-  private Integer port;
-  private Client client;
-
-  public static String QUERY_TYPE = "query";
-  public static String STREAM_TYPE = "stream";
-  public static String TABLE_TYPE = "table";
-
-  public KsqlApiClient(String server, Integer port) {
+  private final URL server;
+  private final Client client;
+
+  public static final String HTTPS = "https";
+  public static final String QUERY_TYPE = "query";
+  public static final String STREAM_TYPE = "stream";
+  public static final String TABLE_TYPE = "table";
+
+  public KsqlApiClient(URL server, @Nullable Configuration.BasicAuth basicAuth) {
+    ClientOptions options = ClientOptions.create()
+        .setHost(server.getHost())
+        .setPort(server.getPort());
+    if (server.getProtocol() != null && server.getProtocol().equals(HTTPS)) {
+      options.setUseTls(true);
+      options.setUseAlpn(true);
```

Comment:
I wonder how this property will behave with old or less up to using old or less up to date TLS support. What do you think?

Reply:
I only tested this on CP 6.0.2 with JDK11 so far - it probably makes sense to provide sensible defaults and allow users to override them with the properties file.
Also an option for setting truststores/keystores might be needed as additional configuration option as mentioned in #287 

These additional properties should then probably end up in `platform.server.ksql.XXX`

Reply:
I love the idea of adding it as config option 👍

Reply:
additional configuration options are now present and documented in `docs/futures/what-ksql-management.rst`